src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -143,6 +166,19 @@ public AzureBlobEgressProvider(ILogger<AzureBlobEgressProvider> logger)
             }
         }
 
+        public void AddConfiguredMetadata(AzureBlobEgressProviderOptions options, EgressArtifactSettings artifactSettings)
+        {
+            if (options.IncludeHostNameAsMetadata)
+            {
+                artifactSettings.Metadata.Add(HostNameMetadataKey, Dns.GetHostName());
+            }
+
+            foreach (var metadataPair in options.Metadata)
+            {
+                artifactSettings.CustomMetadata.Add(metadataPair.Key, Environment.GetEnvironmentVariable(metadataPair.Value));
```

Comment:
This should be getting the environment variable values from the target process, not from the dotnet-monitor process. Use [DiagnosticsClient.GetProcessEnvironmentAsync](https://github.com/dotnet/diagnostics/blob/main/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs#L290). Be mindful that this could be an expensive call, so you'll want to probably make one call to this and use the result throughout the egress invocation.